Release 2.4.1 (#3501)
timescale/timescaledb: CHANGELOG.md

`psql` with the `-X` flag to prevent any `.psqlrc` commands from
accidentally triggering the load of a previous DB version.**

## 2.4.1 (2021-08-19)

> This release contains bug fixes since the 2.4.0 release. We deem it
> high priority to upgrade since it is needed to support PostgreSQL 12.8

Reviewer: ```suggestion
This release contains bug fixes since the 2.4.0 release. 
We deem it high priority to upgrade since it is needed to support PostgreSQL 12.8
```

`psql` with the `-X` flag to prevent any `.psqlrc` commands from
accidentally triggering the load of a previous DB version.**

> ## Unreleased

Reviewer 2: Should the "Unreleased" header be kept in this branch or is it added back later?

Reviewer 3: i think i usually removed it and the first new entry after the release adds it back but no strong opinions either way but in the versioned branch it should be removed

> This release contains bug fixes since the 2.4.0 release.
> 
> We deem it high priority to upgrade since it is needed to support
> PostgreSQL 12.8 and 13.4.

Reviewer 3: ```suggestion
This release contains bug fixes since the 2.4.0 release.
We deem it high priority to upgrade.
The release fixes continous aggregate refresh for postgres 12.8 and 13.4,
a crash with alter table commands and a crash with continuous aggregates with HAVING clause.
```

**Bugfixes**
* #3430 Fix havingqual processing for continuous aggregates
* #3468 Disable tests by default if tools are not found

> * #3468 Fix crash while tracking alter table commands

Reviewer: ```suggestion
* #3462 Fix crash while tracking alter table commands
```

> * #3468 Fix crash while tracking alter table commands

Reviewer 3: wrong PR number

Author: Fixed.

> * #3494 Improve error message when adding data nodes
> * #3498 Fix continuous agg bgw job failure for PG 12.8 and 13.4

Reviewer 4: ```suggestion
* #3489 Fix continuous agg bgw job failure for PG 12.8 and 13.4
* #3494 Improve error message when adding data nodes
```

Author: Thanks for catching it. That is indeed the case.

**Thanks**

> * @brianbenns for reporting a segfault with continuous aggregates

Reviewer 3: ```suggestion
* @brianbenns for reporting a segfault with continuous aggregates
* @usego for reporting an issue with continuous aggregate refresh on PG 13.4
```

See https://github.com/timescale/timescaledb/issues/3496

Author: Done.
